src/test/java/brave/http/ITZipkinHttpClient.java

```diff
@@ -0,0 +1,93 @@
+package brave.http;
+
+import brave.sampler.Sampler;
+import brave.test.http.ITHttpAsyncClient;
+import io.netty.buffer.UnpooledByteBufAllocator;
+import org.junit.AfterClass;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import ratpack.exec.Promise;
+import ratpack.http.client.HttpClient;
+import ratpack.server.ServerConfig;
+import ratpack.test.exec.ExecHarness;
+import ratpack.util.Exceptions;
+import ratpack.zipkin.ClientTracingInterceptor;
+import ratpack.zipkin.internal.DefaultClientTracingInterceptor;
+import ratpack.zipkin.internal.RatpackCurrentTraceContext;
+
+import java.io.IOException;
+import java.net.URI;
+
+public class ITZipkinHttpClient extends ITHttpAsyncClient<HttpClient> {
+
+  private static ExecHarness harness;
+
+  @BeforeClass
+  public static void beforeAll() {
+    harness = ExecHarness.harness();
+  }
+
+  @AfterClass
+  public static void afterAll() {
+    harness.close();
+  }
+
+  @Before public void setup() {
+    Exceptions.uncheck(() -> {
+      harness.run(e -> {
+        currentTraceContext = new RatpackCurrentTraceContext(() -> e);
+        httpTracing = HttpTracing.create(tracingBuilder(Sampler.ALWAYS_SAMPLE).build());
+        client = newClient(server.getPort());
+      });
+    });
+
+  }
+
+  @Override protected HttpClient newClient(int port) {
+    return Exceptions.uncheck(() -> harness.yield(e -> {
+      ClientTracingInterceptor clientTracingInterceptor = new DefaultClientTracingInterceptor(httpTracing, e);
+      return Promise.value(HttpClient.of(s -> s
+        .poolSize(0)
+        .requestIntercept(clientTracingInterceptor::request)
+        .responseIntercept(clientTracingInterceptor::response)
+        .errorIntercept(clientTracingInterceptor::error)
+        .byteBufAllocator(UnpooledByteBufAllocator.DEFAULT)
+        .maxContentLength(ServerConfig.DEFAULT_MAX_CONTENT_LENGTH)));
+    }).getValue());
+  }
+
+  @Override protected void closeClient(HttpClient client) throws IOException {
+    if (client != null) {
+      client.close();
+    }
+  }
+
+  @Override protected void get(HttpClient client, String pathIncludingQuery) throws Exception {
+    harness.yield(e ->
+      client.get(URI.create(url(pathIncludingQuery)))
+    ).getValue();
+    // small delay to ensure the response interceptor is invoked before this method exits.
+    Thread.sleep(1);
+  }
+
+  @Override protected void post(HttpClient client, String pathIncludingQuery, String body) throws Exception {
+    harness.yield(e ->
+      client.post(URI.create(url(pathIncludingQuery)), (request ->
+        request.body(b -> b.text(body))
+      ))
+    ).getValue();
+    // small delay to ensure the response interceptor is invoked before this method exits.
+    Thread.sleep(1);
```

Comment:
I'm not sure why but the response interceptor ran after the blocking get call.  While not a thrilling solution, adding a sleep here fixed the issue.  Since the tests are written in a blocking fashion, and Ratpack code is written in a non blocking fashion, I suspect that actual production code will work as expected.  Again we should investigate testing this more... maybe adding some Ratpack specific tests is warranted here before merging.